kernel/src/hil/eic.rs

```diff
@@ -0,0 +1,55 @@
+//! Interface for external interrupt controller.
+
+/// Enum for selecting which edge to trigger interrupts on.
+#[derive(Debug)]
+pub enum InterruptMode {
+    RisingEdge,
+    FallingEdge,
+    HighLevel,
+    LowLevel,
+}
+
+/// Enum for enabling/disabling filter
+pub enum FilterMode {
+    FilterEnable,
+    FilterDisable,
+}
+
+/// Enum for selecting syn/asyn mode
+pub enum SynchronizationMode {
+    Synchronous,
+    Asynchronous,
+}
+
+/// Interface for EIC.
+pub trait ExternalInterruptController {
+    /// Enables EIC module
+    fn enable(&self);
+
+    /// Disables EIC module
+    fn disable(&self);
+
+    /// Enables external interrupt on line_num
+    /// In asychronous mode, all edge interrupts will be interpreted as level interrupts and the filter is disabled.
+    fn line_enable(&self, line_num: usize);
+
+    /// Disables external interrupt on line_num
+    fn line_disable(&self, line_num: usize);
+
+    /// Configure external interrupt on line_num
+    fn line_configure(
+        &self,
+        line_num: usize,
+        int_mode: InterruptMode,
+        filter: FilterMode,
+        syn_mode: SynchronizationMode,
+    );
```

Comment:
Is there a reason to separate out this function, and not just make it a part of `line_enable()`? Would a user want to enable an external interrupt but not configure it?

Reply:
I've merged enable()/disable() and also configure() into line_enable() function. But now I'm getting a kernel panic "Debug buffer full". In debug.rs, it says I could change the output_buffer size; maybe this was the issue? Any hint would be great!

Reply:
If you try to use `debug!()` too often then it can fill the internal buffer faster than it can actually send the messages over uart and that error is the result. You can either remove some uses of `debug!()` or change the buffer size in kernel/debug.rs.